src/Microsoft.AspNetCore.Mvc.RazorPages/UrlHelperExtensions.cs

```diff
@@ -0,0 +1,103 @@
+            this IUrlHelper urlHelper,
+            string page,
+            object values)
+            => Page(urlHelper, page, values, protocol: null);
+
+        /// <summary>
+        /// Generates a URL with an absolute path for the specified <paramref name="page"/>.
+        /// </summary>
+        /// <param name="urlHelper">The <see cref="IUrlHelper"/>.</param>
+        /// <param name="page">The page to generate the url for.</param>
+        /// <param name="values">An object that contains route values.</param>
+        /// <param name="protocol">The protocol for the URL, such as "http" or "https".</param>
+        /// <returns>The generated URL.</returns>
+        public static string Page(
+            this IUrlHelper urlHelper,
+            string page,
+            object values,
+            string protocol)
+            => Page(urlHelper, page, values, protocol, host: null, fragment: null);
+
+        /// <summary>
+        /// Generates a URL with an absolute path for the specified <paramref name="page"/>.
+        /// </summary>
+        /// <param name="urlHelper">The <see cref="IUrlHelper"/>.</param>
+        /// <param name="page">The page to generate the url for.</param>
+        /// <param name="values">An object that contains route values.</param>
+        /// <param name="protocol">The protocol for the URL, such as "http" or "https".</param>
+        /// <param name="host">The host name for the URL.</param>
+        /// <returns>The generated URL.</returns>
+        public static string Page(
+            this IUrlHelper urlHelper,
+            string page,
+            object values,
+            string protocol,
+            string host)
+            => Page(urlHelper, page, values, protocol, host, fragment: null);
+
+        /// <summary>
+        /// Generates a URL with an absolute path for the specified <paramref name="page"/>.
+        /// </summary>
+        /// <param name="urlHelper">The <see cref="IUrlHelper"/>.</param>
+        /// <param name="page">The page to generate the url for.</param>
+        /// <param name="values">An object that contains route values.</param>
+        /// <param name="protocol">The protocol for the URL, such as "http" or "https".</param>
+        /// <param name="host">The host name for the URL.</param>
+        /// <param name="fragment">The fragment for the URL.</param>
+        /// <returns>The generated URL.</returns>
+        public static string Page(
+            this IUrlHelper urlHelper, 
+            string page, 
+            object values, 
+            string protocol,
+            string host,
+            string fragment)
+        {
+            if (urlHelper == null)
+            {
+                throw new ArgumentNullException(nameof(urlHelper));
+            }
+
+            var routeValues = new RouteValueDictionary(values)
+            {
+                ["page"] = page
+            };
+            return urlHelper.RouteUrl(
```

Comment:
There's some more behavior that's needed here. Basically we need to preserve the value of `page` from ambient values if one wasn't specified. See https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/Routing/UrlHelper.cs#L78